pkg/gitprovider/gogs.go

```diff
@@ -0,0 +1,282 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package gitprovider
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	gogs "github.com/gogs/go-gogs-client"
+)
+
+type GogsGitProvider struct {
+	*AbstractGitProvider
+
+	token      string
+	baseApiUrl string
+}
+
+func NewGogsGitProvider(token string, baseApiUrl string) *GogsGitProvider {
+	provider := &GogsGitProvider{
+		token:               token,
+		baseApiUrl:          baseApiUrl,
+		AbstractGitProvider: &AbstractGitProvider{},
+	}
+	provider.AbstractGitProvider.GitProvider = provider
+
+	return provider
+}
+
+func (g *GogsGitProvider) getApiClient() *gogs.Client {
+	return gogs.NewClient(g.baseApiUrl, g.token)
+}
+
+func (g *GogsGitProvider) CanHandle(repoUrl string) (bool, error) {
+	staticContext, err := g.ParseStaticGitContext(repoUrl)
+	if err != nil {
+		return false, err
+	}
+
+	return strings.Contains(g.baseApiUrl, staticContext.Source), nil
+}
+
+func (g *GogsGitProvider) GetNamespaces(options ListOptions) ([]*GitNamespace, error) {
+	client := g.getApiClient()
+	var namespaces []*GitNamespace
+	user, err := g.GetUser()
+	if err != nil {
+		return nil, err
+	}
+	orgs, err := client.ListMyOrgs()
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch Namespace : %w", err)
+	}
+
+	namespaces = append([]*GitNamespace{{Id: personalNamespaceId, Name: user.Username}}, namespaces...)
+	for _, org := range orgs {
+		namespaces = append(namespaces, &GitNamespace{Id: org.UserName, Name: org.UserName})
+	}
+	return namespaces, nil
+}
+
+func (g *GogsGitProvider) GetUser() (*GitUser, error) {
+	client := g.getApiClient()
+	user, err := client.GetSelfInfo()
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch User : %w", err)
+	}
+	return &GitUser{
```

Comment:
Not 100% sure that this is the right spot to put this but the git config user.name is not set correctly in the workspace.

![Screenshot 2024-10-16 at 13 34 34](https://github.com/user-attachments/assets/1b6196ae-378e-49d6-a203-6a38dbe5f1ff)

Reply:
fixed